pyomo/contrib/gdpopt/branch_and_bound.py

```
@@ -37,442 +45,470 @@
-            heappush(queue, (new_node_data, node_model))  # replace with updated
-                                                          # node data
-        elif node_data.num_unbranched_disjunctions == 0 or \
-             node_data.obj_lb == float('inf'):
-            # We have reached a leaf node, or the best available node is
-            # infeasible.
-            original_model = solve_data.original_model
-            copy_var_list_values(
-                from_list=node_model.GDPopt_utils.variable_list,
-                to_list=original_model.GDPopt_utils.variable_list,
-                config=config,
+            # Make sure disjuncts fixed to False are properly deactivated.
+            for disjunct in disjuncts_fixed_False:
+                disjunct.deactivate()
+
+            # Deactivate nonlinear constraints in unfixed disjuncts
+            for disjunct in unfixed_disjuncts:
+                nonlinear_constraints_in_disjunct = [
+                    constr for constr in disjunct.component_data_objects(
+                        Constraint, active=True)
+                    if constr.body.polynomial_degree() not in _linear_degrees]
+                for constraint in nonlinear_constraints_in_disjunct:
+                    constraint.deactivate()
+                if nonlinear_constraints_in_disjunct:
+                    # TODO might be worthwhile to log number of nonlinear
+                    # constraints in each disjunction for later branching
+                    # purposes
+                    root_util_blk.disjunct_to_nonlinear_constraints[
+                        disjunct] = nonlinear_constraints_in_disjunct
+
+            root_util_blk.disjunction_to_unfixed_disjuncts[
+                disjunction] = unfixed_disjuncts
+            pass
+
+        # Add the BigM suffix if it does not already exist. Used later during
+        # nonlinear constraint activation.
+        # TODO is this still necessary?
```

Would this rewrite answer this question?

---

This I think could be necessary, though probably rarely. The bigm transformation now uses fbbt to calculate M values, so it is successful more often than not now, but this implementation makes sense to me since there is no reason to let that fail (or even run) for a constraint where we know that the `indicator_var` is fixed to `True`.